docs/REFERENCE.md

```diff
@@ -0,0 +1,651 @@
+
+* [Reference](#reference)
+     * [Block Reference](#block-reference)
+          * [Server Block](#server-block)
+          * [Files Block](#files-block)
+          * [SPA Block](#spa-block)
+          * [API Block](#api-block)
+          * [Endpoint Block](#endpoint-block)
+          * [Proxy Block](#proxy-block)
+          * [Request Block](#request-block)
+          * [Response Block](#response-block)
+          * [Backend Block](#backend-block)
+               * [Transport Settings Attributes](#transport-settings-attributes)
+               * [Timings](#timings)
+          * [OpenAPI Block](#openapi-block)
+          * [CORS Block](#cors-block)
+          * [OAuth2 Block](#oauth2-block)
+          * [Definitions Block](#definitions-block)
+          * [Basic Auth Block](#basic-auth-block)
+          * [JWT Block](#jwt-block)
+          * [JWT Signing Profile Block](#jwt-signing-profile-block)
+          * [SAML Block](#saml-block)
+          * [Settings Block](#settings-block)
+     * [Access Control](#access-control)
+     * [Health-Check](#health-check)
+     * [Variables](#variables)
+          * [env](#env)
+          * [request](#request)
+          * [backend_requests](#backend_requests)
+          * [backend_responses](#backend_responses)
+     * [Functions](#functions)
+     * [Modifier](#modifier)
+          * [Request Header](#request-header)
+          * [Response Header](#response-header)
+     * [Parameters](#parameters)
+         * [Query Parameter](#query-parameter)
+         * [Form Parameter](#form-parameter)
+         * [Path Parameter](#path-parameter)
+
+# Reference
+
+## Block Reference 
+
+### Server Block
+
+The `server` block is the main configuration block of Couper's configuration file.
+
+| Block                                | Description |
+|:-------------------------------------|:------------|
+| *context*                            | Root of the configuration file. |
+| *label*                              | &#9888; Mandatory. |
+| **Nested blocks**                    | **Description** |
+| [CORS Block](#cors-block)            | Configures CORS behavior for [Files Block](#files-block), [SPA Block](#spa-block) and [API Block(s)](#api-block) contexts. |
+| [Files Block](#files-block)          | Configures the file serving. |
+| [SPA Block](#spa-block)              | Configures the Web serving for SPA assets. |
+| [API Block(s)](#api-block)           | Configures routing and communication with backend(s). |
+| [Endpoint Block(s)](#endpoint-block) | Configures specific endpoint(s) for current `Server Block` context. |
+| **Attributes**                       | **Description** |
+| `base_path`                          | <ul><li>Optional.</li><li>Configures the path prefix for all requests.</li><li>*Example:* `base_path = "/api"`</li><li>&#9888; Inherited by nested blocks.</li></ul> |
```

Comment:
Should be applied to all html/sub-listings within this document: Replace with proper parent table configuration eg. (block/name, default, required, description).